lib/src/key_value_store/shared_preferences_store.dart

```diff
@@ -45,39 +40,45 @@ class SharedPreferencesStore implements KeyValueStore {
     return val;
   }
 
+  bool _isInt<T>() => _areTypesEqual<T, int>() || _areTypesEqual<T, int?>();
+  bool _isDouble<T>() =>
+      _areTypesEqual<T, double>() || _areTypesEqual<T, double?>();
+  bool _isBool<T>() => _areTypesEqual<T, bool>() || _areTypesEqual<T, bool?>();
+  bool _isString<T>() =>
+      _areTypesEqual<T, String>() || _areTypesEqual<T, String?>();
+
   @override
   Future<void> put<T>(String key, T? value) async {
     _ensureSupportedType<T>();
     final prefs = await _getPrefs();
-
-    if (value == null)
-      prefs.remove(key);
-    else if (value is int)
+  
+    if (value == null) {
+      await prefs.remove(key);
+    } else if (value is int) {
       await prefs.setInt(key, value);
-    else if (value is double)
+    } else if (value is double) {
       await prefs.setDouble(key, value);
-    else if (value is bool)
+    } else if (value is bool) {
       await prefs.setBool(key, value);
-    else if (value is String)
+    } else if (value is String) {
       await prefs.setString(key, value);
-    else if (value is List<String>) await prefs.setStringList(key, value);
+    } else if (value is List<String>) {
+      await prefs.setStringList(key, value);
+    }
   }
 
   void _ensureSupportedType<T>() {
-    if (_areTypesEqual<T, int>() ||
-        _areTypesEqual<T, int?>() ||
-        _areTypesEqual<T, double>() ||
-        _areTypesEqual<T, double?>() ||
-        _areTypesEqual<T, bool>() ||
-        _areTypesEqual<T, bool?>() ||
-        _areTypesEqual<T, String>() ||
-        _areTypesEqual<T, String?>() ||
+    if (_isInt<T>() ||
+        _isDouble<T>() ||
+        _isBool<T>() ||
+        _isString<T>()||
         _areTypesEqual<T, List<String>>() ||
```

Comment:
```suggestion
        _isListOfStrings<T>() ||
```
to match, what do you think?